File: core/inventory/tests/fixtures/sanitizer_cut_corpus_cpp/bypass.c
/* Real bypass — a path reaches the sink without sanitizing.
 *
 * Phase 11 verdict: no_suppress.
 *
 * The else branch passes ``x`` through to render unchanged. Neither
 * lexical nor value-bound should suppress this — it's a real
 * finding.
 */
extern char *g_markup_escape_text(const char *, long);
extern void render(const char *);

void handle(const char *x, int trusted) {
    const char *safe;
    if (trusted) {
        safe = g_markup_escape_text(x, -1);
    } else {
        safe = x;
    }
    render(safe);
}

File: core/inventory/tests/fixtures/sanitizer_cut_corpus/bypass.py
"""Bypass — sanitizer only on one branch; the other reaches the
sink unsanitised.

Phase 4 verdict: no_suppress.

The control-flow cut fails entirely (removing html.escape leaves
the else-branch path intact: entry → cond → else_assign → sink).
This is a real bug that survives to the LLM.
"""


def handle(user):
    if user.is_admin:
        safe = html.escape(user.name)  # noqa: F821 — fixture
    else:
        safe = user.name
    render(safe)                       # noqa: F821 — fixture
